### PR TITLE
Implement extra inline actions on the review page for reviewers and admins

### LINKED ISSUE #7282: Implement "More Actions" section in add-on review pages
Part of #7172

Add a new section to the bottom or add-on review pages (below the private whiteboard), titled "More Actions". It should include the following options:
* "Notify me about new listed versions" toggle option. 
  * This replaces the "Notify me the next time this add-on is updated" checkbox.
  * It also includes a change in how this feature behaves: instead of sending a single update for the next version, the setting should persist until it's toggled off by the reviewer. #7311
* "Clear Admin Code Review Flag" button.  (and same with "Clear Admin Content Review Flag")
  * Present if add-on is flagged for admin review and user has `Reviews:Edit` permission
  * This replaces the "Clear Admin Code Review Flag" checkbox.
* "Listing disabled" toggle option.
  * Present if the user has `Reviews:Edit` permission.
  * This is a new status flag for add-ons. It behaves like the `inactive` (developer disabled) flag, in that the listing will be hidden, but the developer can still access the devhub pages ~~and submit versions~~.

All of these actions should work inline, meaning the user should stay in the add-on review page.

### PR DESCRIPTION
**Pull request by diox**
- Subscribing to notifications
- Force disabling/enabling of an add-on (admins)
- Clear needs admin code/content review flag (admins)

Fix #7282

### CODE REVIEW

File: docs/topics/api/index.rst [hunk 1]
```diff
@@ -41,6 +41,7 @@ using the API.
    discovery
    download_sources
    reviews
+   reviewers
```
[the remark this answers is not in the document]
Comment (eviljeff) [reply docs/topics/api/index.rst]: or perhaps 🙈 💀

File: src/olympia/constants/permissions.py [hunk 2]
```diff
@@ -61,6 +61,9 @@
 # Can moderate add-on ratings submitted by users.
 RATINGS_MODERATE = AclPermission('Ratings', 'Moderate')
 
+# Can access advanced reviewer features.
+REVIEWS_EDIT = AclPermission('Reviews', 'Edit')
```
Comment (eviljeff) [inline src/olympia/constants/permissions.py]: This description is a little vague, and the naming implies they can actually edit reviewer reviews (which I don't think it what it allows)

Comment (diox) [reply src/olympia/constants/permissions.py]: Yeeeah I also find the name confusing. It's from one of the issues but I guess we can change it if we have a better one to suggest?

I really wasn't inspired for the comment, but I figured it was better than nothing :) Also open to ideas there.

Comment (eviljeff) [reply src/olympia/constants/permissions.py]: `Reviews: Admin`?  Still not great, admittedly.

Comment (diox) [reply src/olympia/constants/permissions.py]: 2025e66198edb860e69cde4c5b0028649d237d9e

**Pull request merged**